src/lib/legacy.ts

```diff
@@ -309,8 +309,15 @@ export async function fromV2TargetApps(
 						const appId = parseInt(id, 10);
 						const app = apps[appId];
 
-						// If local mode just use id as uuid
-						const uuid = local ? id : await getUUIDFromAPI(appId);
+						// If local mode or connectivity is not available just use id as uuid
+						const uuid = local
+							? id
+							: await getUUIDFromAPI(appId).catch(() => {
+									throw new Error(
+										// TODO: use the proper CLI version when it is released
+										'Cannot migrate from v2 apps.json without Internet connectivity. Please use a newer CLI with v3 apps.json support.',
```

Comment:
I see the balena-cli PR bumping balena-preload to include https://github.com/balena-io-modules/balena-preload/pull/273 isn't out yet -- would it be better to include a specific version of the CLI in this error message, once the CLI PR goes through?

Reply:
Otherwise LGTM!

Reply:
I think so, hence the TODO :). `balena-preload` was just merged, I'll bump the CLI

Reply:
Ahh sorry, I didn't see that 😅

Reply:
@cywang117 updated the message

Reply:
@pipex Looks good! Feel free to self-certify.